app/src/main/java/ai/elimu/appstore/ui/applications/ApplicationListActivity.java

```diff
@@ -65,13 +65,21 @@ protected void onStart() {
         RoomDb.databaseWriteExecutor.execute(() -> {
             List<Application> applications = applicationDao.loadAll();
             Log.d(getClass().getName(), "applications.size(): " + applications.size());
-            applicationListAdapter.setApplications(applications);
-
+            if(applications != null) {
+                applicationListAdapter.setApplications(applications);
+            }
```

Comment:
```suggestion
```

This code change is not necessary, right? (Since `applicationListAdapter.notifyDataSetChanged();` is the line causing the error.)